openssl/src/cms.rs

```diff
@@ -80,4 +91,43 @@ impl CmsContentInfo {
             Ok(CmsContentInfo::from_ptr(cms))
         }
     }
+
+    /// Given a signing cert `signcert`, private key `pkey`, a certificate stack `certs`,
+    /// data `data` and flags `flags`, create a CmsContentInfo struct.
+    ///
+    /// All arguments are optional.
+    ///
+    /// OpenSSL documentation at [`CMS_sign`]
+    ///
+    /// [`CMS_sign`]: https://www.openssl.org/docs/manmaster/man3/CMS_sign.html
+    pub fn sign<T: HasPrivate>(
+        signcert: Option<&X509>,
+        pkey: Option<&PKeyRef<T>>,
+        certs: Option<&Stack<X509>>,
+        data: Option<&[u8]>,
+        flags: u32,
```

Comment:
Still needs a `bitflags!` type for the flags.

Reply:
On it.